projects/components/src/utils/test/locator/widget-locator.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { Type } from '@angular/core';
+
+/**
+ * A class that is able to make queries to the DOM and be instantiated by a widget finder.
+ *
+ * All widget locators should extend this base class.
+ *
+ * @example
+ *
+ * class LoginWidgetLocator<T> extends BaseWidgetLocator<T> {
+ *      getUsernameField(): T {
+ *          return this.driver.get('.username').unwrap();
+ *      }
+ *
+ *      getOkButton(): T {
+ *          return this.driver.getByText('button', 'Ok').unwrap();
+ *      }
+ *
+ *      getOkButtonContainer(): T {
+ *          return this.driver.getByText('button', 'Ok').parent('.button-container').unwrap();
+ *      }
+ * }
+ */
+export class BaseWidgetLocator<T> {
```

Comment:
The base class is a widget object, not a locator, it uses a locator. 

I think the base object should just be `WidgetObject`

Reply:
Talked as a team and the new suggestion is `WidgetElementLocator`. Please make sure documentation specifies that it should not try to bundle actions, such as click, type, into the class.